### DISCUSSION TITLE
User authentication

### DISCUSSION THREAD

**Discussion by andriusmv** [shuding/nextra#444] [Ideas]
What would be the best way to implement user authentication/authorization? Next AuthJS (https://next-auth.js.org/) sounds like a good one.

This way users could access only certain parts of the documents template.

**Comment by rulim34** [comment]
Hi, this is a nice feature to have.

Do you already figured it out how to add auth on Nextra docs site? I want to try implement it myself too.

**Comment by crguezl** [comment]
I have written a tutorial about this. Is in https://ull-mii-sytws.github.io/next-auth-getting-started/

**Comment by dharanigowthamsampath** [comment]
with clerk we can implement that. its simple as easy

**Comment by phucbm** [comment]
Nextra with authentication is possible. I created a private docs site for my team using [Supabase auth](https://supabase.com/docs/guides/auth).

**Reply by phucbm** [reply]
Guys! It would be great if this is a built-in feature of Nextra, while we're waiting for that, what are your suggestions to integrate authentication with Nextra? I might have some time to create a template repository for this. 

I have some options:
- [Supabase auth](https://supabase.com/docs/guides/auth): I already had a working version of private Nextra with supabase.
- Authjs: https://authjs.dev/
- etc.
